examples/apps/ai-collab/src/components/UserPresenceGroup.tsx

```diff
@@ -0,0 +1,137 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+"use client";
+
+import { InteractiveBrowserCredential } from "@azure/identity";
+import { Client } from "@microsoft/microsoft-graph-client";
+// eslint-disable-next-line import/no-internal-modules
+import { TokenCredentialAuthenticationProvider } from "@microsoft/microsoft-graph-client/authProviders/azureTokenCredentials";
+import { Avatar, Badge, styled } from "@mui/material";
+import React, { useEffect, useState } from "react";
+import { v4 as uuid } from "uuid";
+
+import type { UserPresence } from "@/app/presence";
+
+interface UserPresenceProps {
+	userPresenceGroup: UserPresence;
+}
+
+const UserPresenceGroup: React.FC<UserPresenceProps> = ({
+	userPresenceGroup,
+}): JSX.Element => {
+	const [photoUrls, setPhotoUrls] = useState<string[]>([]);
+	const [isPhotoFetched, setIsPhotoFetched] = useState<boolean>(false);
+
+	// this effect will run when the userPresenceGroup changes and will update the photoUrls state
+	useEffect(() => {
+		const allPhotos: string[] = [];
+		for (const element of [...userPresenceGroup.onlineUsers.clientValues()]) {
+			for (const user of [...element.items.values()]) {
+				allPhotos.push(user.value.value.photo);
+			}
+		}
+		setPhotoUrls(allPhotos);
+	}, [userPresenceGroup]);
+
+	/**
+	 * this effect will run once when the component mounts and will fetch the user's photo if it's spe client,
+	 * for the tinylicious client, it will use the default photo.
+	 * */
+	useEffect(() => {
+		const fetchPhoto = async (): Promise<void> => {
+			const clientId = process.env.NEXT_PUBLIC_SPE_CLIENT_ID;
+			const tenantId = process.env.NEXT_PUBLIC_SPE_ENTRA_TENANT_ID;
+			if (tenantId === undefined || clientId === undefined) {
+				// add a default photo for tinylicious client
+				userPresenceGroup.onlineUsers.local.set(`id-${uuid()}`, {
```

Comment:
For tinylicious client, the default photo should be added into container presence group here for each opening new tab. And it should trigger the previous effect function to re-render the page like the spe client does.

Reply:
Is it happening for you, though? As you can see in my screenshot, I only see one circle with default photo, but I have 2 tabs open.

Reply:
Seems like it didn't re-render the page for you. I can see tinylicious clients on my local. Let me see how to make it refresh page itself...
![Screenshot (8)](https://github.com/user-attachments/assets/b6c407a7-8af6-4d15-be84-a93bd6078372)

Reply:
More than re-rendering the page, I think the presence group should update real-time as things (the connected clients) change. The presence framework probably has events to tell us when things change?

I'm still curious though, I just started the app from the latest commit in the PR (https://github.com/microsoft/FluidFramework/pull/22850/commits/dd433b17d1f1f416e15d869347add5b40fa25b4f) and I'm still only seeing one circle, even when I start the second client from a different browser 🤔 .

Reply:
If you click refresh on page manually, the second circle should show-up. It was running slow to appear on my side too. I put the `userPresenceGroup` in state but it seems not detect the object changed and failed to re-render the page auto. I guess it does need a handler to monitor the `addUser` event and refresh page.